deepspeed/runtime/zero/stage3.py

```diff
@@ -350,7 +350,7 @@ def fetch_sub_module(self, current_submodule: Module) -> None:
                     params_to_fetch))
             while self.__param_queue and len(discarded_from_prefetch_queue) < len(
                     params_not_already_fetched):
-                param_in_trace = self.__param_queue.popleft()
```

Comment:
Are you seeing some failure with this code? Please take a look at this [description](https://github.com/microsoft/DeepSpeed/blob/0eb2c763b47344ecdacb88713a710dc2753da39f/deepspeed/runtime/zero/stage3.py#L217-L219).

Reply:
Thanks @tjruwase - yes I am seeing a bug introduced in #1453 that triggers this [Runtime Error](https://github.com/microsoft/DeepSpeed/blob/0eb2c763b47344ecdacb88713a710dc2753da39f/deepspeed/runtime/zero/stage3.py#L358) for some models